consensus/src/data_manager.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+// use futures::channel::{mpsc, mpsc::Sender, oneshot};
+use crate::quorum_store::batch_reader::BatchReader;
+use crate::quorum_store::types::Data;
+use aptos_types::transaction::SignedTransaction;
+use arc_swap::ArcSwapOption;
+use consensus_types::proof_of_store::{LogicalTime, ProofOfStore};
+use tokio::sync::oneshot;
+
+/// Notification of execution committed logical time for QuorumStore to clean.
+#[async_trait::async_trait]
+pub trait DataManager: Send + Sync {
+    /// Notification of committed logical time
+    async fn notify_commit(&self, logical_time: LogicalTime);
+
+    fn new_epoch(&self, data_reader: Arc<BatchReader>);
+
+    async fn get_data(&self, poss: Vec<ProofOfStore>) -> Vec<SignedTransaction>;
```

Comment:
This can take a Payload. That way we don't have to expose any implementation details of the Payload in the trait.

Reply:
Good. Will do.